apps/next-product-starter/src/pages/video/[slug].tsx

```diff
@@ -233,47 +238,64 @@ const VideoCuesList: React.FC<any> = () => {
           note = {text: cue.text}
         }
         return (
-          <VideoCue key={note.text} cue={cue} isActive={isActive} note={note} />
+          <VideoCue
+            key={note.text + i}
+            cue={cue}
+            isActive={isActive}
+            note={note}
+          />
         )
       })}
     </ul>
   )
 }
 
-const Page = ({data}: any) => {
+const Page = ({data, nextLesson}: any) => {
+  const router = useRouter()
+
   return (
-    <Layout>
+    <main className="">
       <VideoProvider
         services={{
           addCueNote,
           loadViewer:
             (context: VideoStateContext, _event: VideoEvent) => async () => {
-              context.viewer = VIEWER[1] as any
+              context.viewer = VIEWER[0] as any
             },
         }}
         actions={{
-          onVideoEnded: (_context: any, _event: any) => {
-            console.log('do stuff')
+          onVideoEnded: (context: VideoStateContext, event: VideoEvent) => {
+            // if autoplay is on and we have nextLesson, navigate to next lesson
+            if (nextLesson && context.autoplay) {
+              router.push(`/video/${nextLesson.slug}`)
+            }
```

Comment:
the "autoplay" here is "autoplay next lesson in the playlist" and NOT `autoplay` in terms of HTMLVideoElement which is used for autoplaying videos on page load and distinctly problematic in this context

**the video player context shouldn't have the concept of autoplay in state**

autoplaying the next lesson in the playlist is the concern of the video page, not the player